### DISCUSSION TITLE
Frida （ Accessing android underlayers）

### DISCUSSION THREAD

**Discussion by yst-one** [flet-dev/flet#3814] [Packages] [answered]
Frida （ Accessing android underlayers）

**Comment by ndonkoHenri** [comment, marked as answer]
Frida can't run inside a Flet mobile app — this is fundamental to what Frida is, not a packaging gap:

- Frida is a process-injection / instrumentation toolkit. The Python module is a controller that attaches to other processes via a frida-server daemon (needs root on Android) or a jailbroken iOS device.
- A normal sandboxed app can't `ptrace`/`task_for_pid` or spawn-and-inject other processes — the OS sandbox forbids it. So even a perfectly built wheel would be a dead import in a Flet app.
- (It also pulls a huge native build — V8, a forked Vala compiler, GLib, etc.)

Your title mentions "accessing Android underlayers" — for that, the supported approach is [pyjnius](https://flet.dev/blog/tap-into-native-android-and-ios-apis-with-Pyjnius-and-pyobjus/), which lets you call Android Java/system APIs directly from Python. That's the right tool for accessing platform internals from a Flet app.

**Reply by yst-one** [reply]
Sorry, I didn't know much about this library back then. I'm closing this request now, and thank you for your hard work.
